lib/vnet/ssh_handler.go

```diff
@@ -92,37 +96,121 @@ func (h *sshHandler) handleTCPConnectorWithTargetConn(
 	}
 	defer localConn.Close()
 
-	// For now we accept the incoming SSH connection but forwarding to the
-	// target is not implemented yet so we immediately close it.
+	var (
+		preAuthConn      ssh.ServerPreAuthConn
+		clientConn       *sshConn
+		clientConnErr    error
+		initiatedSSHConn bool
+	)
+	sendAuthBanner := func(msg string) {
+		// This is super hacky but:
+		// - Auth banners need to be sent before the authentication phase
+		//   completes.
+		// - As soon as we return from PublicKeyCallback, x/crypto/ssh will
+		//   complete the authentication phase.
+		// - Therefore we need to call SendAuthBanner before returning
+		//   from PublicKeyCallback.
+		// - However, SendAuthBanner will block until the authentication phase
+		//   completes so we must call it from a goroutine.
```

Comment:
Are you sure about this? It seems bizarre and sort of counter to the intended use of `ssh.ServerPreAuthConn.SendAuthBanner`.

Reply:
not sure how i came to these conclusions, deleted